src/AstSemanticChecker.cpp

```diff
@@ -247,8 +247,8 @@ void AstSemanticChecker::checkProgram(AstTranslationUnit& translationUnit) {
             return;
         }
 
-        for (size_t i = 0; i < fun.getArity(); i++) {
-            auto arg = fun.getArg(i);
+        for (size_t i = 0; i < fun.getArguments().size(); i++) {
```

Comment:
can you change this loop to size_t i=0; for(auto arg: fun.getArguments()) { ....; i++}